src/app/public/modules/inline-form/inline-form.component.ts

```diff
@@ -0,0 +1,268 @@
+import {
+  ChangeDetectorRef,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnInit,
+  OnDestroy,
+  Output,
+  TemplateRef
+} from '@angular/core';
+
+import 'rxjs/add/observable/zip';
+
+import 'rxjs/add/operator/take';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  SkyAppWindowRef
+} from '@skyux/core';
+
+import {
+  SkyLibResourcesService
+} from '@skyux/i18n';
+
+import {
+  skySlideDissolve
+} from './animations/slide-dissolve';
+
+import {
+  SkyInlineFormAdapterService
+} from './inline-form-adapter.service';
+
+import {
+  SkyInlineFormCloseArgs
+} from './types/inline-form-close-args';
+
+import {
+  SkyInlineFormConfig
+} from './types/inline-form-config';
+
+import {
+  SkyInlineFormButtonConfig
+} from './types/inline-form-button-config';
+
+import {
+  SkyInlineFormButtonLayout
+} from './types/inline-form-button-layout';
+
+@Component({
+  selector: 'sky-inline-form',
+  templateUrl: './inline-form.component.html',
+  styleUrls: ['./inline-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [ skySlideDissolve ]
+})
+export class SkyInlineFormComponent implements OnInit, OnDestroy {
+
+  @Input()
+  public config: SkyInlineFormConfig;
+
+  @Input()
+  public template: TemplateRef<any>;
+
+  @Input()
+  public set showForm(value: boolean) {
+    this._showForm = value;
+
+    if (value) {
+      this.skyAppWindowRef.nativeWindow.setTimeout(() => {
```

Comment:
Maybe add a comment above this `setTimeout` to explain why it exists? Helpful for future debugging...

Reply:
Great idea. Done.